The 1.19.3 Update

ChampionAsh5357 opened the pull request:
Updates the documentation to conform with the current state of 1.19.3 in Forge. This also fixes some inconsistencies in the original documentation and updates external links to their new locations.

I expect there will probably be some changes when MinecraftForge/MinecraftForge#9182, MinecraftForge/MinecraftForge#9185, MinecraftForge/MinecraftForge#9188 get merged in some capacity, so if those are merged before this, I will update.

ChampionAsh5357 committed:
Update to 1.19.3

ChampionAsh5357 committed:
Update to MinecraftForge/MinecraftForge#9198

sciwhiz12 commented on docs/concepts/registries.md:
```diff
@@ -53,10 +53,10 @@ public void register(RegisterEvent event) {
 
 !!! important
-    Dynamic registry objects should only be registered in-code if an existing vanilla registry object uses it.
+    Dynamic registry objects can **only** be registered through data files (e.g. JSON). It **cannot** be registered in-code.
```
Subject pronoun in second sentence (`it`) should follow the plurality of its antecedent (`Dynamic registry objects`).
```suggestion
    Dynamic registry objects can **only** be registered through data files (e.g. JSON). They **cannot** be registered in-code.
```

sciwhiz12 commented on docs/datagen/server/advancements.md:
````diff
@@ -1,9 +1,43 @@
+An `AdvancementSubProvider` is responsible for generating advancements, containing a method which takes in a registry lookup and the writer (`Consumer<Advancement>`). A subtype of `AdvancementSubProvider` should take in an `ExistingFileHelper` to use while building an `Advancement`.
+
+```java
+// In some AdvancementSubProvider
````
Shouldn't this be `In some MyAdvancementSubProvider`?

ChampionAsh5357 replied:
That is outdated, we've moved to using `ForgeAdvancementProvider` and `AdvancementGenerator` (which should be in the most recent commit of the docs). Additionally, it could be, but it would be incorrect to assume all types are of `MyAdvancementSubProvider` when they could be subtypes of `AdvancementSubProvider`.

sciwhiz12 replied:
Whoops, I meant `In MyAdvancementSubProvider`, because the constructor calls for the class's name to be `MyAdvancementSubProvider`. I think I had this review comment from a previous attempt to review, before the newer commits.

ChampionAsh5357 committed:
Update to new PRs

- MinecraftForge/MinecraftForge#9170
- MinecraftForge/MinecraftForge#9182
- MinecraftForge/MinecraftForge#9188

ChampionAsh5357 committed:
Fix incorrectness in model provider

ChampionAsh5357 committed:
Remove keyboard handler reference

sciwhiz12 commented on docs/items/index.md:
```diff
@@ -10,23 +10,66 @@ Creating an Item
+
+### Custom Creative Tabs
+
+A custom `CreativeModeTab` can be created via `CreativeModeTabEvent$Register#registerCreativeModeTab` on the [mod event bus][modbus]. This takes in the name of the tab and a consumer of the builder. In addition, a list of `ResourceLocation`s or `CreativeModeTab`s can be provided to determine where this tab should be located.
```
> In addition, a list of `ResourceLocation`s or `CreativeModeTab`s can be provided to determine where this tab should be located.

Should we give an example here?

ChampionAsh5357 replied:
In my opinion, I think an example on one of the methods is fine since this would be passing in two lists with resource locations, something everybody should know how to construct and get the correct method signature within the event. But, I'll leave it up to your discretion on which method should be showcased in the docs.

sciwhiz12 replied:
Hmmm. Given that most mods are likely to append a tab to the end instead of ordering based on other tabs, we can omit the example then.

sciwhiz12 commented on docs/gui/screens.md:
```diff
@@ -55,9 +55,9 @@ The offset can be obtained by calling `#getBlitOffset` and set using `#setBlitOf
+## Renderable
 
-`Widget`s are essentially objects that are rendered. These include screens, buttons, chat boxes, lists, etc. `Widget`s only have one method: `#render`. This takes in the `PoseStack` holding any prior transformations to properly render the widget, the x and y positions of the mouse scaled to the relative screen size, and the tick delta (how many ticks have passed since the last frame).
+`Renderable`s are essentially objects that are rendered. These include screens, buttons, chat boxes, lists, etc. `Renderable`s only have one method: `#render`. This takes in the `PoseStack` holding any prior transformations to properly render the widget, the x and y positions of the mouse scaled to the relative screen size, and the tick delta (how many ticks have passed since the last frame).
```
This paragraph contains the first reference to the concept of a "widget". In the past, since the interface was named `Widget`, it was fine. However, because it is now `Renderable`, it may be preferrable to introduce the concept of a "widget" and explain how it relates to `Renderable`.

sciwhiz12 commented on docs/resources/server/recipes/ingredients.md:
````diff
@@ -30,9 +30,9 @@ Though they are functionally identical, Compound ingredients replaces the way on
 ```
 
-#### NBTIngredient
+#### StrictNBTIngredient
````
There is another reference to `NBTIngredient` in the `PartialNBTIngredient` section that needs to be corrected.

sciwhiz12 commented on docs/misc/gametest.md:
```diff
@@ -247,6 +247,19 @@ property 'forge.enabledGameTestNamespaces', 'modid1,modid2,modid3'
 The Game Test Server is a special configuration which runs a build server. The build server returns an exit code of the number of required, failed Game Tests. All failed tests, whether required or optional, are logged. This server can be run using `gradlew runGameTestServer`.
 
+!!! important
+    Due to a quirk in how Gradle works, by default, if a task forces a system exit before it has finished executing, even in a zero code exit, the task will be marked as the failure. ForgeGradle sets by default a force exit on run tasks such that any subprojects are not executed in sequence. However, as such, the Game Test Server will always fail.
```
Note that the actual behavior is that the task action runs `System.exit` which _kills the Gradle daemon_, causing the Gradle runner to report a build failure.

ChampionAsh5357 replied:
I've changed the wording on this to hopefully reflect that point.

sciwhiz12 commented on docs/misc/updatechecker.md:
```diff
@@ -48,10 +48,11 @@ You can retrieve the results of the Forge Update Checker using `VersionChecker#g
-| `BETA_OUTDATED` | There is a new unstable version. |
-|          `BETA` | The current version is equal to or newer than the latest unstable version. |
+|    `UP_TO_DATE` | The current version is equal to the recommended version. |
+|    `AHEAD` | The current version is newer than the recommended version if there is not latest version. |
```
This entry should be aligned with the first column.
```suggestion
|         `AHEAD` | The current version is newer than the recommended version if there is not a latest version available. |
```

sciwhiz12 commented on docs/datagen/client/modelproviders.md:
```diff
@@ -47,7 +66,7 @@ Finally, the model can set whether to use ambient occlusion in a level (`ModelBu
 ### `BlockModelBuilder`
 
-A `BlockModelBuilder` represents a block model to-be-generated. There is no additional functionality compared to a `ModelBuilder`.
+A `BlockModelBuilder` represents a block model to-be-generated. In Addition to the `ModelBuilder`, a transform to the entire model (`BlockModelBuilder#rootTransform`) can be generated. The root can be translated (`RootTransformBuilder#transform`), rotated (`RootTransformBuilder#rotation`, `RootTransformBuilder#postRotation`), and scaled (`RootTransformBuilder#scale`) either individually or all in ...
```
Typo: `Addition` -> `addition`.

```suggestion
A `BlockModelBuilder` represents a block model to-be-generated. In addition to the `ModelBuilder`, a transform to the entire model (`BlockModelBuilder#rootTransform`) can be generated. The root can be translated (`RootTransformBuilder#transform`), rotated (`RootTransformBuilder#rotation`, `RootTransformBuilder#postRotation`), and scaled (`RootTransformBuilder#scale`) either individually or all in one transformation (`RootTransformBuilder#transform`) around some origin (`RootTransformBuilder#origin`).
```

sciwhiz12 commented on docs/datagen/server/advancements.md:
```diff
@@ -1,9 +1,42 @@
+`ForgeAdvancementProvider$AdvancementGenerator`
+-----------------------------------------------
+
+An `ForgeAdvancementProvider$AdvancementGenerator` is responsible for generating advancements, containing a method which takes in a registry lookup, the writer (`Consumer<Advancement>`), and the existing file helper..
```
Typo: `An` -> `A`.
```suggestion
A `ForgeAdvancementProvider$AdvancementGenerator` is responsible for generating advancements, containing a method which takes in a registry lookup, the writer (`Consumer<Advancement>`), and the existing file helper..
```

sciwhiz12 commented on docs/datagen/server/advancements.md:
````diff
@@ -1,9 +1,42 @@
+An `ForgeAdvancementProvider$AdvancementGenerator` is responsible for generating advancements, containing a method which takes in a registry lookup, the writer (`Consumer<Advancement>`), and the existing file helper..
+
+```java
+// In some ForgeAdvancementProvider$AdvancementGenerator or as a lambda reference
````
This should note "in some subclass of"
```suggestion
// In some subclass of ForgeAdvancementProvider$AdvancementGenerator or as a lambda reference
```

ChampionAsh5357 committed:
Address comments

sciwhiz12 commented on docs/gui/screens.md:
```diff
@@ -57,9 +57,9 @@ The offset can be obtained by calling `#getBlitOffset` and set using `#setBlitOf
+`Renderable`s are essentially objects that are rendered. These include screens, buttons, chat boxes, lists, etc. `Renderable`s only have one method: `#render`. This takes in the `PoseStack` holding any prior transformations to properly render the renderable, the x and y positions of the mouse scaled to the relative screen size, and the tick delta (how many ticks have passed since the last frame).
 
-Some common widgets which may be used are `Button`, its subtype `ImageButton`, and `EditBox` which is used to input text on the screen.
+Some common renderables are screens and 'widgets': interactable elements which typically render on the screen  such as `Button`, its subtype `ImageButton`, and `EditBox` which is used to input text on the screen.
```
Accidental doubled space.
```suggestion
Some common renderables are screens and 'widgets': interactable elements which typically render on the screen such as `Button`, its subtype `ImageButton`, and `EditBox` which is used to input text on the screen.
```

ChampionAsh5357 committed:
Things I miss while reading

Co-authored-by: sciwhiz12 <sciwhiz12@gmail.com>

sciwhiz12 reviewed (approved):
:shipit:

Shadows-of-Fire commented:
This will probably also introduce changes that may be worth noting on docs: https://github.com/MinecraftForge/MinecraftForge/pull/9106
